Sources/_StringProcessing/Engine/Processor.swift

```diff
@@ -411,9 +426,9 @@ extension Processor {
 
     case .consumeBy:
       let reg = payload.consumer
-      guard currentPosition < bounds.upperBound,
+      guard currentPosition < subjectBounds.upperBound,
```

Comment:
Subject or search bounds?

Reply:
How do we, generally, know which one to choose?

Reply:
Generally, we want to be moving and matching with `searchBounds`; `subjectBounds` should really only be used to match anchors, not for general purpose index movements.

Reply:
(i.e. this was incorrect)

Reply:
Any way we could use access control or some better coding convention here? E.g. if we always want search bounds except for anchors, can we isolate the code that refers to subject bounds inside the engine?